arch/arm/src/arm/Toolchain.defs

```diff
@@ -153,7 +155,6 @@ ifeq ($(CONFIG_LTO_FULL),y)
     NM := $(CROSSDEV)gcc-nm
     ARCHOPTIMIZATION += -fuse-linker-plugin
     ARCHOPTIMIZATION += -fno-builtin
-    ARCHOPTIMIZATION += -nodefaultlibs
```

Comment:
I'm not sure if that matters, but with this change we add `-nostartfiles` and `-nodefaultlibs` unconditionally for ARM.
I hope that it should not matter, just want to be sure that all is under control

Reply:
since we provide the start files, and link the libgcc etc explicitly, it should be fine to always link with -nostdlib(-nostartfiles and -nodefaultlibs).

Reply:
Yeah. I mean that it is strange that we didn't have it will now.

Reply:
Should -nostdlib be applied to all targets that use gcc for compilation, at least avr uses avr_gcc ? I have also seen that some avr builds pull definitions from libgcc which is definitely wrong.

Reply:
Done.